sorcery.lic

```diff
@@ -47,22 +60,28 @@ class Sorcery
     true
   end
 
-  def focus_runestone
+  def focus_runestone(target_mindstate)
     return if DRSkill.getrank('Sorcery') > 50
-
-    case bput('get my runestone', 'You get', 'I could not find', 'What were you referring to')
+    case bput('get my runestone', 'You get', 'I could not find', 'What were you referring to','You are already holding that')
     when 'I could not find', 'What were you referring to'
       return
     end
-
-    5.times do
-      break if DRSkill.getxp('Sorcery') >= 30
-
-      bput('focus my runestone', 'You focus your magical senses')
+    maxfocuses = 5
+    maxmindstate = 30
+    if target_mindstate && target_mindstate.to_i <= 34
+      maxfocuses = 1000
+      maxmindstate = target_mindstate.to_i
+    end
+    maxfocuses.times do
+      break if DRSkill.getxp('Sorcery') >= maxmindstate 
+       bput('focus my runestone', 'You focus your magical senses')
       waitrt?
     end
 
-    bput('stow runestone', 'You put your')
+   
+      bput('stow runestone', 'You put your')
+
+    return
```

Comment:
Relic of a less elegant way I tried initially, combined with not really being a Ruby person. It should go.